text/0000-precise-pre-release-dependencies.md

```diff
@@ -0,0 +1,139 @@
+| `a.b.c-pre0`    | `a.b.c-pre0`       | `x.y.z`        | ❌²                       | ✅                                  |
+
+✅: Will upgrade
+
+❌: Will not upgrade
+
+¹For backwards compatibility with Cargo's current behaviour
+
+²Emits a warning
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Pre-release versions are not easily auditable when they are only specified in the lock file.
+  A change that makes use of a pre-release version may not be noticed during code review as reviewers don't always check for changes in the lock file.
+- Library crates that require a pre-release version are not well supported (see [future-possibilities])
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The main alternative to this would be to accept that pre-release versions are not very usable and discourage their use.
+Cargo overrides can be used instead using `[patch]`.
+These provide a similar experience to pre-releases, however, they require that the library's code is somehow vendored outside of the registry, usually with git.
+This can cause issues particularly in CI where jobs may have permission to fetch from a private registry but not from private git repositories.
+Resolving issues around not being able to fetch pre-releases from the registry usually wastes a significant amount of time.
+
+Another alternative would be to resolve pre-release versions in `Cargo.toml`s even when a dependency specifies a stable version.
+This is explored in [future-possibilities].
+This would require significant changes to the resolver since the latest compatible version would depend on the versions required by other parts of the dependency tree.
+For that reason, I consider detailing such a change outside of the scope of this RFC.
+
+# Prior art
+[prior-art]: #prior-art
+
+[RFC: Precise Pre-release Deps](https://github.com/rust-lang/rfcs/pull/3263) aims to solve a similar but subtly different issue where `cargo update` opts to upgrade 
+pre-release versions to new pre-releases when one is released.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+It would be nice if dependencies could specify their requirements for pre-release versions.
+
+Take for example this dependency tree.
+
+```
+example
+├── a ^0.1.0
+│   └── b =0.1.1-pre0
+└── b ^0.1.0
+```
+
+Since crates ignore the lock files of their dependacies there is no way for `a` to comunicate with `example` that it requires features from `b = 0.1.1-pre0` without breaking `example`s direct dependacy on `b`.
+To enable this we could extend the use of the concept of compatible pre-releases to work in `Cargo.toml`.
+This would require that pre-releases are specified with `=` and would allow pre-release versions to be requested anywhere within the dependacy tree without causing the resolver to thow an error.
```

Comment:
```suggestion
This would require that pre-releases are specified with `=` and would allow pre-release versions to be requested anywhere within the dependency tree without causing the resolver to throw an error.
```